Session/Conversations/ConversationViewModel.swift

```diff
@@ -395,11 +396,11 @@ public class ConversationViewModel: OWSAudioPlayerDelegate {
         /// write queue when it isn't needed, in order to do this we:
         ///
         ///   - Don't bother marking anything as read if there are no unread interactions (we can rely on the
-        ///     `threadData.threadUnreadCount` to always be accurate)
+        ///     `threadData.threadHasUnreadMessagesOfAnyKind` to always be accurate)
         ///   - Don't bother marking anything as read if this was called with the same `interactionId` that we
         ///     previously marked as read (ie. when scrolling and the last message hasn't changed)
         guard
-            (self.threadData.threadUnreadCount ?? 0) > 0,
+            self.threadData.threadHasUnreadMessagesOfAnyKind == true,
```

Comment:
I'm concerned that this is a little misleading because the logic in `Interaction` will actively prevent a bunch of types from even being unread - is there a reason for adding a new `threadHasUnreadMessagesOfAnyKind` instead of just updating the old `threadUnreadCount` logic?

(It'd also be good to avoid adding the new value since it requires a second `SUM` which means it needs to do an aggregate calculation on the database table again)

Reply:
This a tricky one, I feel it is a little misleading too. `threadUnreadCount` only counts incoming messages and incoming calls, but with the latest design of disappearing messages, we need to mark some info messages as read too. So the problem comes and I need to use this `threadHasUnreadMessagesOfAnyKind` to indicate all kinds of messages that can be unread, so I can mark those info messages as read.

Reply:
Ah I understand, that's a bit frustrating lol

Might be good to update the `Interaction.wasRead` property description to something like the below so the different behaviours are documented
```
/// A flag indicating whether the interaction has been read (this is a flag rather than a timestamp because
/// we couldn’t know if a read timestamp is accurate)
///
/// This flag is used:
///  - In conjunction with `Interaction.variantsToIncrementUnreadCount` to determine the unread count for a thread
///  - In order to determine whether the "Disappear After Read" expiration type should be started
```